core/model/src/driver.rs

```diff
@@ -236,3 +236,92 @@ impl RpcMessage for ValidateAllocation {
     type Item = bool;
     type Error = GenericError;
 }
+
+// ************************** ENTER **************************
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Enter {
+    amount: String,
+    network: Option<String>,
+    token: Option<String>,
+}
+
+impl Enter {
+    pub fn new(amount: String, network: Option<String>, token: Option<String>) -> Enter {
```

Comment:
Parsing amount should be done in CLI, not in driver.
```suggestion
    pub fn new(amount: BigDecimal, network: Option<String>, token: Option<String>) -> Enter {
```